tests/capture/autograph/test_if_else.py

```diff
@@ -0,0 +1,308 @@
+
+check_cache = TRANSFORMER.has_cache
+
+
+@pytest.fixture(autouse=True)
+def enable_disable_plxpr():
+    qml.capture.enable()
+    yield
+    qml.capture.disable()
+
+
+class TestConditionals:
+    """Test that the autograph transformations produce correct results on conditionals."""
+
+    def test_simple_cond(self):
+        """Test basic function with conditional."""
+
+        def circuit(n):
+            if n > 4:
+                res = n**2
+            else:
+                res = n
+
+            return res
+
+        # can't convert to jaxpr without autorgraph
+        with pytest.raises(
+            jax.errors.TracerBoolConversionError,
+            match="Attempted boolean conversion of traced array",
+        ):
+            jax.make_jaxpr(circuit)(1)
+
+        # with autograph we can convert to jaxpr
+        circuit = run_autograph(circuit)
+        jaxpr = jax.make_jaxpr(circuit)(0)
+        assert "cond" in str(jaxpr)
+
+        def res(x):
+            return eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, x)
+
+        # evaluating the jaxpr gives expected results
+        assert res(0) == [0]
+        assert res(1) == [1]
+        assert res(2) == [2]
+        assert res(3) == [3]
+        assert res(4) == [4]
+        assert res(5) == [25]
+        assert res(6) == [36]
+
+    def test_cond_one_else_if(self):
+        """Test a cond with one else_if branch"""
+
+        def circuit(x):
+            if x > 2.7:
+                res = x * 4
+            elif x > 1.4:
+                res = x * 2
+            else:
+                res = x
+
+            return res
+
+        ag_circuit = run_autograph(circuit)
+        jaxpr = jax.make_jaxpr(ag_circuit)(0)
+        assert "cond" in str(jaxpr)
+
+        def res(x):
+            return eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, x)[0]
+
+        assert res(4) == 16
+        assert res(2) == 4
+        assert res(1) == 1
+
+    def test_cond_many_else_if(self):
+        """Test a cond with multiple else_if branches"""
+
+        def circuit(x):
+            if x > 4.8:
+                res = x * 8
+            elif x > 2.7:
+                res = x * 4
+            elif x > 1.4:
+                res = x * 2
+            else:
+                res = x
+
+            return res
+
+        ag_circuit = run_autograph(circuit)
+        jaxpr = jax.make_jaxpr(ag_circuit)(0)
+        assert "cond" in str(jaxpr)
+
+        def res(x):
+            return eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, x)[0]
+
+        assert res(5) == 40
+        assert res(3) == 12
+        assert res(2) == 4
+        assert res(-3) == -3
+
+    def test_qubit_manipulation_cond(self):
+        """Test conditional with quantum operation."""
+
+        @qml.qnode(qml.device("default.qubit", wires=1))
+        def circuit(x):
+            if x > 4:
+                qml.PauliX(wires=0)
+
+            m = measure(wires=0)
+
+            return qml.expval(m)
```

Comment:
Note that these tests will fail when we port to natively executing the jaxpr on the device.  Since this will be merged first, I'll just mark them xfail in mine.